### ISSUE TITLE
Move usernames out of message bubbles

### ISSUE THREAD

**Issue by kentnek** [flyerhq/flutter_chat_ui#370]
Hi, currently username is inside the message bubble. How can I move it outside, to the top of the bubble?

![image](https://user-images.githubusercontent.com/7024160/208493053-b4d237a6-bc72-45a4-8993-cd311260c51b.png)

Thanks for the great library!

**Comment by AmirC123** [comment]
+1 
It's a good idea because it makes it so you could customize bubble color & text color to fit just each other, rather than having them having to fit all possible group user colors

BUT it should be customizable. User name, message body, and bubble ideally should each be customizable separately for both layout and style.

Since we already have all the message builders and a customBubbleBuilder, this could possibly be achieved by having a user name builder, and an additional field in the customBubbleBuilder

**Comment by demchenkoalex** [comment]
Why user name builder will be required exactly? `bubbleBuilder` is what you need with a `showUserNames` set to false? Impossible to make options for everything, everyone wants something different :( I think those high lever builders should be used to make something you need in your case

**Comment by kentnek** [comment]
> Why user name builder will be required exactly? `bubbleBuilder` is what you need with a `showUserNames` set to false? Impossible to make options for everything, everyone wants something different :( I think those high lever builders should be used to make something you need in your case

I thinking exposing `showUserName` to `bubbleBuilder` is good enough for my case, since only the topmost bubble in a group of messages needs to show the name. In fact, as we already have `nextMessageInGroup`, how about having another param `isFirstMessageInGroup`?

**Comment by demchenkoalex** [comment]
I don't think it is possible, I can't go back and see if this was a first message, chat is not performant as is like people mention here, imagine more calculations :D

**Comment by kentnek** [comment]
@demchenkoalex then just exposing `showUserName` to `bubbleBuilder` should work I guess.

**Comment by demchenkoalex** [comment]
this I can do
